drivers/ethernet/phy/phy_mii.c

```diff
@@ -205,6 +205,14 @@ static int update_link_state(const struct device *dev)
 
 		k_sleep(K_MSEC(100));
 
+		/* On some PHY chips, the BMSR bits are latched, so the first read may
+		 * show incorrect status. A second read ensures correct values.
+		 */
+		if (phy_mii_reg_read(dev, MII_BMSR, &bmsr_reg) < 0) {
```

Comment:
generally don't like to see phy specific code in the generic driver but this is small enough I don't think it's worth making a fuss over